internal/recipient/recipient.go

```diff
@@ -33,6 +33,24 @@ func (r Key) MarshalText() (text []byte, err error) {
 	}
 }
 
+// CopyNonNil copies non-nil fields from the provided recipient.Key to the current one and
+// returns the modified recipient key.
+func (r Key) CopyNonNil(recipientKey Key) Key {
```

Comment:
This function and how you use `recipient.Key` in `incident.HistoryRow` contradicts the current assumption in the code that there's a 1:1 relation between `recipient.Recipient` and `recipient.Key`, as it can be seen in multiple places:

https://github.com/Icinga/icinga-notifications/blob/002d14dc19e6a68bb894d0c9ad8829463c68cc9b/internal/recipient/recipient.go#L16-L21
https://github.com/Icinga/icinga-notifications/blob/002d14dc19e6a68bb894d0c9ad8829463c68cc9b/internal/recipient/recipient.go#L23-L34
https://github.com/Icinga/icinga-notifications/blob/002d14dc19e6a68bb894d0c9ad8829463c68cc9b/internal/config/runtime.go#L92-L112

I'd want to keep that assumption in place as `recipient.Key` is used as a key in maps for example and that would result in nasty bugs if `recipient.Key` values with more than one member set would ever be used in a map key.

So I think it would be better to add the three columns as explicit members in `incident.HistoryRow` if it's now intended that more than one is set to a non-null value.